src/game/Tactical/Soldier_Control.cc

```diff
@@ -9085,6 +9215,751 @@ static void SetSoldierPersonalLightLevel(SOLDIERTYPE* const s)
+		}
+	}
+
+	if(apSoldier.empty()) return;
+
+	bool changed = true;
+
+	while(changed)
+	{
+		changed = false;
+
+		for(unsigned int i = 1; i < apSoldier.size(); ++i)
+		{
+			if(aSCR[i - 1] < aSCR[i])
+			{
+				std::swap(aSCR[i - 1], aSCR[i]);
+				std::swap(apSoldier[i - 1], apSoldier[i]);
+				changed = true;
+			}
+		}
+	}
+
+	for(unsigned int i = 0; i < apSoldier.size(); ++i)
+	{
+		SoldierEquipFromLoadedSector(apSoldier[i], full_access);
+	}
+}
+
+void TeamDropAll(UINT8 bTeam, UINT8 bSide)
+{
+	FOR_EACH_SOLDIER(s)
+	{
+		if(s->bTeam == bTeam && s->bSide == bSide && s->ubProfile == NO_PROFILE) SoldierDropAllWithAnimation(s, false);
+	}
+}
+
+bool hasVehicleInventory(SOLDIERTYPE const* const s)
+{
+	return (GCM->getGamePolicy()->vehicle_inventory && IsMechanical(*s) && (s->ubProfile==PROF_HUMMER || s->ubProfile==PROF_ICECREAM || s->ubProfile==PROF_ELDERODO));
+}
+
+void SoldierDropFromSlot(SOLDIERTYPE* const pSoldier, UINT8 const slot, bool const stack, bool const cleanup)
+{
+	if(!pSoldier) return;
+	if(slot>=NUM_INV_SLOTS) return;
+	if(pSoldier->inv[slot].usItem == NOTHING) return;
+
+	Visibility const visibility = pSoldier->bSide == OUR_TEAM ? VISIBLE : INVISIBLE;
+
+	bool const here = pSoldier->sSectorX == gWorldSectorX && pSoldier->sSectorY == gWorldSectorY && pSoldier->bSectorZ == gbWorldSectorZ;
+
+	GridNo sGridNo = pSoldier->sGridNo;
+
+	OBJECTTYPE item;
+
+	if(!stack)
+	{
+		ItemFromStackRemoveBest(&pSoldier->inv[slot], &item);
+	}
+	else
+	{
+		item=pSoldier->inv[slot];
+		DeleteObj(&pSoldier->inv[slot]);
+	}
+
+	if(!here) AddItemsToUnLoadedSector(pSoldier->sSectorX, pSoldier->sSectorY, pSoldier->bSectorZ, sGridNo, 1, &item, pSoldier->bLevel, WOLRD_ITEM_FIND_SWEETSPOT_FROM_GRIDNO | WORLD_ITEM_REACHABLE, 0, visibility);
+	else AddItemToPool(pSoldier->sGridNo, &item, visibility, pSoldier->bLevel, 0, -1);
+
+	if(cleanup)
+	{
+		DirtyMercPanelInterface(pSoldier, DIRTYLEVEL2);
+		fInterfacePanelDirty = DIRTYLEVEL2;
+		HandleAllReachAbleItemsInTheSector( pSoldier->sSectorX,  pSoldier->sSectorY,  pSoldier->bSectorZ );
+	}
+}
+
+void SoldierDropAll(SOLDIERTYPE* const pSoldier, bool const cleanup)
+{
+	if(!pSoldier) return;
+
+	Visibility const visibility = pSoldier->bSide == OUR_TEAM ? VISIBLE : INVISIBLE;
+
+	bool const here = pSoldier->sSectorX == gWorldSectorX && pSoldier->sSectorY == gWorldSectorY && pSoldier->bSectorZ == gbWorldSectorZ;
+
+	if(!here)
+	{
+		HandleLeavingOfEquipmentInCurrentSector(*pSoldier);
+		SoldierWipeInventory(pSoldier);
+		return;
+	}
+
+	if(pSoldier->bSide == OUR_TEAM)
+	{
+		for(int i=0; i<NUM_INV_SLOTS; ++i)
+		{
+			if(pSoldier->inv[i].usItem==NOTHING) continue;
+			AddItemToPool(pSoldier->sGridNo, &pSoldier->inv[i], visibility, pSoldier->bLevel, 0, -1);
+			DeleteObj(&pSoldier->inv[i]);
+		}
+	}
+	else //WORKAROUND for enemy soldiers use sector equipment
+	{
+		for(int i=0; i<NUM_INV_SLOTS; ++i)
+		{
+			switch(pSoldier->inv[i].usItem)
+			{
+				case KEY_1 ... KEY_32:
```

Comment:
This is what breaks the build ... since it appears to be a non-standard gcc extension.

Reply:
Thank you looking into the error, I have pushed an update (squashed into IME) in an attempt to solve this problem.